### PR TITLE
Add missing `isNull()` implementation to `FilteredAggregator`

### PR DESCRIPTION
**Pull request by abhishekrb19**
This PR adds an `isNull()` implementation to `FilteredAggregator`. Otherwise, the default implementation always returns
false, which can be wrong, and will cause jobs to fail when invoking accessors like `getDouble()`, `getFloat()` on the delegate when the underlying data can contain null:

```
java.lang.RuntimeException: java.lang.IllegalStateException: Cannot return float for Null Value
	at org.apache.druid.indexing.seekablestream.SeekableStreamIndexTaskRunner.runInternal(SeekableStreamIndexTaskRunner.java:605) ~[druid-indexing-service]
	at org.apache.druid.indexing.seekablestream.SeekableStreamIndexTaskRunner.run(SeekableStreamIndexTaskRunner.java:266) ~[druid-indexing-service]
	at org.apache.druid.indexing.seekablestream.SeekableStreamIndexTask.runTask(SeekableStreamIndexTask.java:151) ~[druid-indexing-service]
	at org.apache.druid.indexing.common.task.AbstractTask.run(AbstractTask.java:173) ~[druid-indexing-service]
	at org.apache.druid.indexing.overlord.SingleTaskBackgroundRunner$SingleTaskBackgroundRunnerCallable.call(SingleTaskBackgroundRunner.java:477) ~[druid-indexing-service]
	at org.apache.druid.indexing.overlord.SingleTaskBackgroundRunner$SingleTaskBackgroundRunnerCallable.call(SingleTaskBackgroundRunner.java:449) ~[druid-indexing-service]
	at java.util.concurrent.FutureTask.run(FutureTask.java:264) ~[?:?]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128) ~[?:?]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628) ~[?:?]
	at java.lang.Thread.run(Thread.java:829) ~[?:?]
Caused by: java.lang.IllegalStateException: Cannot return float for Null Value
	at org.apache.druid.query.aggregation.NullableNumericAggregator.getFloat(NullableNumericAggregator.java:83) ~[druid-processing]
	at org.apache.druid.query.aggregation.FilteredAggregator.getFloat(FilteredAggregator.java:52) ~[druid-processing]
	at org.apache.druid.segment.incremental.OnheapIncrementalIndex.getMetricHelper(OnheapIncrementalIndex.java:578) ~[druid-processing]
	at org.apache.druid.segment.incremental.OnheapIncrementalIndex.getMetricFloatValue(OnheapIncrementalIndex.java:461) ~[druid-processing]
	at org.apache.druid.segment.incremental.IncrementalIndex$FloatMetricColumnSelector.getFloat(IncrementalIndex.java:1481) ~[druid-processing]
```

Test class updates:
- `FilteredAggregatorTest` uses `TestNullableFloatColumnSelector` instead of the simple aggregator to catch any null handling-related issues.
- General cleanup of the files as functions were scattered everywhere in the file, making it hard to follow.
- `FilteredAggregatorTest` and `WrappingDimensionSelectorTest` directly extend `InitializedNullHandlingTest`. The static null handling init blocks initialized test helper classes are removed (which caught me by surprise when testing as things were initialized twice)

Perhaps in a separate change, we can migrate some of the tests that still use simple column selectors to the nullable ones since they're more versatile.

<hr>

##### Key changed/added classes in this PR
 * `FilteredAggregator.java`
 * `FiilteredAggregatorTest.java`

<hr>

This PR has:

- [x] been self-reviewed.
   - [ ] using the [concurrency checklist](https://github.com/apache/druid/blob/master/dev/code-review/concurrency.md) (Remove this item if the PR doesn't have any relation to concurrency.)
- [ ] added documentation for new or modified features or behaviors.
- [ ] a release note entry in the PR description.
- [ ] added Javadocs for most classes and all non-trivial methods. Linked related entities via Javadoc links.
- [ ] added or updated version, license, or notice information in [licenses.yaml](https://github.com/apache/druid/blob/master/dev/license.md)
- [ ] added comments explaining the "why" and the intent of the code wherever would not be obvious for an unfamiliar reader.
- [x] added unit tests or modified existing tests to cover new code paths, ensuring the threshold for [code coverage](https://github.com/apache/druid/blob/master/dev/code-review/code-coverage.md) is met.
- [ ] added integration tests.
- [x] been tested in a test Druid cluster.

### CODE REVIEW

**Review by pranavbhole — APPROVED** [review]

File: processing/src/test/java/org/apache/druid/query/aggregation/TestFloatNullableColumnSelector.java [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.druid.query.aggregation;
+
+
+public class TestFloatNullableColumnSelector extends TestFloatColumnSelector
```
Comment (clintropolis) [inline processing/src/test/java/org/apache/druid/query/aggregation/TestFloatNullableColumnSelector.java:23]: nit: is this really worth being in its own file instead of just like.. overriding something inline or a static class in the test file?

File: processing/src/test/java/org/apache/druid/query/aggregation/FilteredAggregatorTest.java [hunk 2]
```diff
@@ -84,6 +85,40 @@ public void testAggregate()
...
+        new SelectorDimFilter("dim", "a", null)
+    );
+
+    final FilteredAggregator agg = (FilteredAggregator) factory.factorize(makeColumnSelector(selector));
+    aggregate(selector, agg);
+    assertValues(agg, selector);
+  }
+
+  @Test
+  public void testAggregateWithNullableSelectorAndNonNullValues()
```
Comment (clintropolis) [inline processing/src/test/java/org/apache/druid/query/aggregation/FilteredAggregatorTest.java:106]: nit: since isNull on the selector always returns true is this really testing null and non-null values?

File: processing/src/test/java/org/apache/druid/query/aggregation/FilteredAggregatorTest.java [hunk 3]
```diff
@@ -218,14 +253,22 @@ public ColumnCapabilities getColumnCapabilities(String columnName)
 
   private void assertValues(FilteredAggregator agg, TestFloatColumnSelector selector, double... expectedVals)
   {
-    Assert.assertEquals(NullHandling.defaultDoubleValue(), agg.get());
-    Assert.assertEquals(NullHandling.defaultDoubleValue(), agg.get());
-    Assert.assertEquals(NullHandling.defaultDoubleValue(), agg.get());
-    for (double expectedVal : expectedVals) {
```
Comment (clintropolis) [inline processing/src/test/java/org/apache/druid/query/aggregation/FilteredAggregatorTest.java:224]: maybe this should accept `Double...` instead of `double...` so that it confirms that it _actually_ tests the expected values, since in this case the selectory always returns null it seems like for the test cases using `TestFloatNullableColumnSelector` this `!agg.isNull` branch will never be hit...

**Review by clintropolis — COMMENTED** [review]
seems like an oversight this was left out of #6233

File: processing/src/test/java/org/apache/druid/query/aggregation/TestFloatNullableColumnSelector.java (hunk 1 above)
Comment (abhishekrb19) [reply processing/src/test/java/org/apache/druid/query/aggregation/TestFloatNullableColumnSelector.java:23]: yeah, I agree this was a quick dirty hack. I remembered this test class a while back: https://github.com/apache/druid/blob/master/processing/src/test/java/org/apache/druid/segment/TestNullableFloatColumnSelector.java. I can cleanup this test to use `TestNullableFloatColumnSelector` which takes in `Float` and determines `isNull()` on the fly as opposed to returning a static value.

**Review by zachjsh — APPROVED** [review]
LGTM

**Review by clintropolis — APPROVED** [review]
